libobs/obs-encoder.c

```diff
@@ -371,6 +365,8 @@ static void remove_connection(struct obs_encoder *encoder, bool shutdown)
 	 * up again */
 	if (shutdown)
 		obs_encoder_shutdown(encoder);
+	encoder->initialized = false;
+
 	set_encoder_active(encoder, false);
```

Comment:
@tt2468:
> I can assure you that moving `encoder->initialized` in the first commit changes no behavior. It was just worth doing to keep the code concise.

the order of `encoder->initialized = false` and `set_encoder_active(encoder, false)` is swapped with this change; I don't currently see if this change is significant, however

arguably the old behavior was probably wrong, since e.g. the call chain around `full_stop` didn't include any memory barriers after `encoder->initialized = false;`?

this is probably fine in the grand scheme of things, and I also mostly prefer that this is fixed in libobs, rather than requiring [libobs-user changes](https://github.com/obsproject/obs-studio/pull/10890) (this* is already part of Twitch beta `30.2.0-beta4-enhanced-broadcasting-v27` see https://github.com/amazon-contributing/upstreaming-to-obs-studio/commits/release/v27/, and we haven't gottan any issue reports related to this)

Reply:
Resolving per off-thread discussion and because merging requires conversations to be closed.